plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftConnector.java

```diff
@@ -0,0 +1,760 @@
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.ConfigBuilder;
+import io.fabric8.openshift.client.DefaultOpenShiftClient;
+import io.fabric8.openshift.client.OpenShiftClient;
+
+/**
+ * Client for OpenShift API.
+ *
+ * @author ML
+ */
+@Singleton
+public class OpenShiftConnector extends DockerConnector {
+    private static final Logger LOG = LoggerFactory.getLogger(OpenShiftConnector.class);
+    private static final String CHE_CONTAINER_IDENTIFIER_LABEL_KEY = "cheContainerIdentifier";
+    private static final String CHE_DEFAULT_EXTERNAL_ADDRESS = "172.17.0.1";
+    private static final String CHE_OPENSHIFT_RESOURCES_PREFIX = "che-ws-";
+    private static final String CHE_WORKSPACE_ID_ENV_VAR = "CHE_WORKSPACE_ID";
+    private static final int CHE_WORKSPACE_AGENT_PORT = 4401;
+    private static final int CHE_TERMINAL_AGENT_PORT = 4411;
+    private static final String DOCKER_PREFIX = "docker://";
+    private static final String DOCKER_PROTOCOL_PORT_DELIMITER = "/";
+    private static final String OPENSHIFT_SERVICE_TYPE_NODE_PORT = "NodePort";
+    private static final int OPENSHIFT_LIVENESS_PROBE_DELAY = 120;
+    private static final int OPENSHIFT_LIVENESS_PROBE_TIMEOUT = 1;
+    private static final int OPENSHIFT_WAIT_POD_DELAY = 1000;
+    private static final int OPENSHIFT_WAIT_POD_TIMEOUT = 120;
+    private static final String OPENSHIFT_POD_STATUS_RUNNING = "Running";
+    private static final String OPENSHIFT_DEPLOYMENT_LABEL = "deployment";
+    private static final String OPENSHIFT_IMAGE_PULL_POLICY_IFNOTPRESENT = "IfNotPresent";
+    private static final Long UID_ROOT = Long.valueOf(0);
+    private static final Long UID_USER = Long.valueOf(1000);
+
+    private final OpenShiftClient    openShiftClient;
+    private final KubernetesLabelConverter kubernetesLabelConverter;
+    private final KubernetesEnvVar kubernetesEnvVar;
+    private final KubernetesContainer kubernetesContainer;
+    private final KubernetesService kubernetesService;
+    private final String             cheOpenShiftProjectName;
+    private final String             cheOpenShiftServiceAccount;
+
+    @Inject
+    public OpenShiftConnector(DockerConnectorConfiguration connectorConfiguration,
+                              DockerConnectionFactory connectionFactory,
+                              DockerRegistryAuthResolver authResolver,
+                              DockerApiVersionPathPrefixProvider dockerApiVersionPathPrefixProvider,
+                              @Named("che.openshift.endpoint") String openShiftApiEndpoint,
+                              @Named("che.openshift.username") String openShiftUserName,
+                              @Named("che.openshift.password") String openShiftUserPassword,
+                              @Named("che.openshift.project") String cheOpenShiftProjectName,
+                              @Named("che.openshift.serviceaccountname") String cheOpenShiftServiceAccount) {
```

Comment:
is there a reason why some of attributes for che.* are starting with che and others by openShift ?